server/asset/eth/eth.go

```diff
@@ -461,24 +642,22 @@ func (eth *Backend) ValidateSignature(addr string, pubkey, msg, sig []byte) erro
 // poll pulls the best hash from an eth node and compares that to a stored
 // hash. If the same does nothing. If different, updates the stored hash and
 // notifies listeners on block chans.
-func (eth *Backend) poll(ctx context.Context) {
+func (eth *AssetBackend) poll(ctx context.Context) {
 	best := &eth.bestHash
 	send := func(reorg bool, err error) {
 		if err != nil {
 			eth.log.Error(err)
 		}
-		eth.blockChansMtx.RLock()
-		for c := range eth.blockChans {
-			select {
-			case c <- &asset.BlockUpdate{
-				Reorg: reorg,
-				Err:   err,
-			}:
-			default:
-				eth.log.Error("failed to send block update on blocking channel")
-			}
+		u := &asset.BlockUpdate{
+			Reorg: reorg,
+			Err:   err,
+		}
+
+		eth.sendBlockUpdate(u)
+
+		for _, be := range eth.tokens {
+			be.sendBlockUpdate(u)
 		}
```

Comment:
> Does `eth.tokens` also need a lock, or are all the tokens already expected to be loaded by now?

They are all expected to be loaded. https://github.com/decred/dcrdex/blob/105d3e730f5ac193e8fe91c30fb4615a18b260e5/server/dex/dex.go#L526-L541

> Also, could this potentially take a long time? Maybe depends on how many tokens we have and how many markets. Is it one send per coin per market it is used in?
> 
> Polling interval is one second. I don't know if we may just miss polling intervals if this method `poll` takes too long. That's probably fine though?

So I'm not **too** worried about it taking too long here. Maybe I can run in a goroutine? But you've reminded me of something I've been meaning to get into. I think we should meter block updates from all assets, since `Swapper` does a `processBlock` and a `checkInactionBlockBased` for every block. Ethereum blocks are coming every 10 seconds on average, so there'll be a ton of relatively simultaneous blocks. I think `Swapper` could use some kind of metering mechanism, ensuring that there is at least say 5 or 10 seconds between taking action on sequential block checks.

Reply:
+1 to metering in Swapper

Reply:
> Maybe I can run in a goroutine?

I was gonna suggest this, but we definitely wan the updates to happen sequentially. I don't think a goroutine would be good.